ctrd/client_opts.go

```diff
@@ -0,0 +1,82 @@
+package ctrd
+
+type clientOpts struct {
+	startDaemon            bool
+	debugLog               bool
+	rpcAddr                string
```

Comment:
Should we have default value for the option? I'm wondering that the `WithXXX` is not necessary for every time. WDYT?

Reply:
Not call `WithXXX` will set the `Opts` with default value.

Reply:
Hi @HusterWan 

```
// set default value for parameters
	if copts.rpcAddr == "" {
		copts.rpcAddr = unixSocketPath
	}
	client.rpcAddr = copts.rpcAddr

	if copts.grpcClientPoolCapacity <= 0 {
		copts.grpcClientPoolCapacity = defaultGrpcClientPoolCapacity
	}
	if copts.maxStreamsClient <= 0 {
		copts.maxStreamsClient = defaultMaxStreamsClient
	}
```

For this part, I means we can use the following code to handle

```
var defaultClientOpts = clientOpts{
rpcAddr: unixSocketPath,
grpcClientPoolCapacity: defaultGrpcClientPoolCapacity,
maxStreamsClient: defaultMaxStreamsClient
}
```

With the default value, we don't need to have the if-else code. :)

Hope it can help

Reply:
I am afraid, i can not agree with you, `default` value take effect when not specified or invalid params